mono/metadata/custom-attrs.c

```diff
@@ -566,21 +569,27 @@ MONO_RESTORE_WARNING
 		case MONO_TYPE_CLASS:
 		case MONO_TYPE_OBJECT:
 		case MONO_TYPE_STRING:
-		case MONO_TYPE_SZARRAY:
+		case MONO_TYPE_SZARRAY: {
+			HANDLE_FUNCTION_ENTER ();
+			MONO_HANDLE_NEW (MonoArray, arr);
+
 			for (i = 0; i < alen; i++) {
 				MonoObject *item = NULL;
 				load_cattr_value (image, m_class_get_byval_arg (tklass), &item, p, boundp, &p, error);
 				if (!is_ok (error))
 					return NULL;
```

Comment:
should call HANDL_FUNCTION_RETURN() befor return?

Reply:
It does look wrong, agreed, but your suggestion seems wrong too.
@vargaz Make it single-point-of-return, please. Well. Not exactly, but you understand.